bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -351,9 +355,31 @@ void ReplicaImp::removeDuplicatedRequestsFromRequestsQueue() {
   while (first != nullptr && !clientsManager->canBecomePending(first->clientProxyId(), first->requestSeqNum())) {
     primaryCombinedReqSize -= first->size();
     delete first;
-    requestsQueueOfPrimary.pop();
+    requestsQueueOfPrimary.pop_front();
     first = (!requestsQueueOfPrimary.empty() ? requestsQueueOfPrimary.front() : nullptr);
   }
+
+  LOG_INFO(GL, "" << KVLOG(requestsQueueOfPrimary.size()));
+
+  struct ClientsReqCmparator {
+    bool operator()(const ClientRequestMsg *const lhs, const ClientRequestMsg *const rhs) {
+      return (lhs->clientProxyId() < rhs->clientProxyId() ||
```

Comment:
why not just <=?

Reply:
The std::set requires us to implement strict lower comparison on the data type, I think. We compare first the **clientProxyId**-s and if those are equal we further check the **requestSeqNum**-s.